boards/arm/nucleo_h723zg/doc/index.rst

```diff
@@ -0,0 +1,245 @@
++-----------+------------+-------------------------------------+
+| ETHERNET  | on-chip    | ethernet                            |
++-----------+------------+-------------------------------------+
+| RNG       | on-chip    | True Random number generator        |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not yet supported on this Zephyr port.
+
+The default configuration per core can be found in the defconfig files:
+``boards/arm/nucleo_h723zg/nucleo_h723zg_defconfig``
+
+For mode details please refer to `STM32 Nucleo-144 board User Manual`_.
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+The Nucleo H723ZG board features a ST Zio connector (extended Arduino Uno V3)
+and a ST morpho connector. Board is configured as follows:
+
+- UART_3 TX/RX : PD8/PD9 (ST-Link Virtual Port Com)
+- USER_PB : PC13
+- LD1 : PB0
+- LD2 : PB7
+- LD3 : PB14
+- I2C : PB8, PB9
+
+System Clock
+------------
+
+Nucleo H723ZG System Clock could be driven by an internal or external
+oscillator, as well as the main PLL clock. By default, the System clock is
+driven by the PLL clock at 550MHz, driven by an 8MHz high-speed external clock.
+
+Serial Port
+-----------
+
+Nucleo H723ZG board has 4 UARTs and 4 USARTs. The Zephyr console output is
+assigned to UART3. Default settings are 115200 8N1.
+
+Programming and Debugging
+*************************
+
+Currently the ``nucleo_h723zg`` board supports only OpenOCD and J-Link (not tested) debugger.
+
+.. note::
+
+   Official OpenOCD support for this board was added on October '20.
+   Make sure your openocd version is older than that.
+   Following links may be helpful: `OpenOCD installing Debug Version`_
+   and `OpenOCD installing with ST-LINK V3 support`_
+   You can also choose the ``stm32cubeprogrammer`` runner.
+
+.. note::
+
+   Check if your ST-LINK V3 has newest FW version. It can be done with `STM32CubeIDE`_
+
+Flashing
+========
+
+Nucleo H723ZG board includes an ST-LINK/V3 embedded debug tool interface.
+
+First, connect the NUCLEO-H723ZG to your host computer using
+the USB port to prepare it for flashing. Then build and flash your application.
+
+Here is an example for the :ref:`hello_world` application.
+
+Run a serial host program to connect with your NUCLEO-H723ZG board.
+
+.. code-block:: console
+
+   $ minicom -b 115200 -D /dev/ttyACM0
+
+or use scrreen:
+
+.. code-block:: console
+
+   $ screen /dev/ttyACM0 115200
+
+Build and flash the application:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: nucleo_h723zg
+   :goals: build flash
+
+You should see the following message on the console:
+
+.. code-block:: console
+
+   $ Hello World! nucleo_h723zg
+
+Blinky example can also be used:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/basic/blinky
+   :board: nucleo_h723zg
+   :goals: build flash
+
+.. note::
+
+   Flashing the board and pushing RESTART button on the board leads
```

Comment:
Is it required to push restart button when using cube programmer tool?

Reply:
It is not required. Removed this notice.

Reply:
It is not. Removed the notice